src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -102,4 +87,49 @@ public int size() {
     public boolean isEmpty() {
         return size == 0;
     }
+
+    private void rangeCheckForAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException(outOfBoundsMsg(index));
+        }
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException(outOfBoundsMsg(index));
+        }
+    }
+
+    private String outOfBoundsMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
+
+    private int newLength(int oldLength, int minGrowth, int prefGrowth) {
+        return oldLength + Math.max(minGrowth, prefGrowth);
+    }
+
+    private Object[] grow() {
+        return grow(size + 1);
+    }
```

Comment:
I did not understand why you have 2 methods, this method should increase the size of the array

Reply:
grow() is called a couple times in my program, but if there were no such method, I had to use the grow(int minSize) method and pass size + 1 as an argument to it in all the calls to it. So the grow() method is just a simplification to not write unnecessary code

Reply:
Thanks for the explanation, I didn't understand it right away